rct229/rulesets/ashrae9012019/section11/section11rule15.py

```diff
@@ -0,0 +1,213 @@
+                    ) != swh_use_p.get("entering_water_mains_temperature_schedule"):
+                        rule_status = "fail"
+                        rule_note = (
+                            rule_note
+                            + swh_use_b["id"]
+                            + " Service Water Heating Distribution System entering main water temperature schedules do not match. "
+                        )
+                    swh_dist_sys_id_b = swh_use_b.get("served_by_distribution_system")
+                    swh_dist_sys_id_p = swh_use_p.get("served_by_distribution_system")
+                    if swh_dist_sys_id_b != swh_dist_sys_id_p:
+                        rule_status = "fail"
+                        rule_note = (
+                            rule_note
+                            + swh_use_b["id"]
+                            + " Service water heating distribution system that serves this water heating use units are "
+                            "inconsistent between proposed and baseline models. "
+                        )
+                    # do we require swh_dist_sys_id_p and swh_dist_sys_id_b not None?
+                    if not swh_dist_sys_id_b and swh_dist_sys_id_b == swh_dist_sys_id_p:
+                        swh_dist_sys_b = swh_dist_sys_dict_b[swh_dist_sys_id_b]
+                        swh_dist_sys_p = swh_dist_sys_dict_p[swh_dist_sys_id_p]
+                        if swh_dist_sys_b.get(
+                            "design_supply_water_temperature"
+                        ) != swh_dist_sys_p.get("design_supply_water_temperature"):
+                            rule_status = "fail"
+                            rule_note = (
+                                rule_note
+                                + swh_use_b["id"]
+                                + " Service Water Heating Distribution System design water supply temperatures do not "
+                                "match. "
+                            )
+                    if rule_status == "pass":
+                        if swh_use_p.get("use", 0) < swh_use_b.get("use", 0):
+                            rule_status = "undetermined"
+                            manual_check_msg = (
+                                manual_check_msg
+                                + swh_use_b["id"]
+                                + " Proposed Service Water Heating Use is less than the baseline.  Manually verify that reduction is due to an ECM that reduces service water heating use, such as low-flow fixtures. "
+                            )
+                        elif swh_use_p.get("use", 0) > swh_use_b.get("use", 0):
+                            rule_status = "fail"
+                            rule_note = (
+                                rule_note
+                                + swh_use_b["id"]
+                                + " Proposed Service Water Heating Use is greater than the baseline. "
+                            )
+                return {
```

Comment:
I am not in favor of strictly following the RDS implementation. - `calc_val` function suppose to do calculations and return values for reporting and rule_check/manual_check functions. In here, the rule definition calculation, these are all blended into `calc_val` function.

I suggest focusing on return a few parameters:

```python
return {
    is_swh_use_none_b: swh_use_b is None,
    is_swh_use_none_p: swh_use_p is None,
    swh_use_use_units_b: swh_use_b.get("use_units"),
    swh_use_use_units_p: swh_use_b.get("use_units"),
    swh_use_multiplier_schedule_b: swh_use_b.get("use_multiplier_schedule"),
    swh_use_multiplier_schedule_p: swh_use_p.get("use_multiplier_schedule"),
    swh_use_temperature_at_fixture_b: swh_use_b.get("temperature_at_fixture"),
    swh_use_temperature_at_fixture_p: swh_use_p.get("temperature_at_fixture"),
    swh_use_water_mains_temperature_schedule_b: swh_use_b.get("entering_water_mains_temperature_schedule"),
    swh_use_water_mains_temperature_schedule_p: swh_use_p.get("entering_water_mains_temperature_schedule"),
    swh_use_served_by_distribution_system_b: swh_dist_sys_id_b = swh_use_b.get("served_by_distribution_system"),
    swh_use_served_by_distribution_system_p: swh_dist_sys_id_p = swh_use_b.get("served_by_distribution_system"),
    swh_use_design_supply_water_temperature_b: design_supply_water_temperature_b, # you will need to calculate this value
    swh_use_design_supply_water_temperature_p: design_supply_water_temperature_p, # you will need to calculate this value
    swh_use_use_b: swh_use_b.get("use", 0),
    swh_use_use_p: swh_use_p.get("use", 0)
}
```


Move all the logics including rule_note draft to the rule_check function.
Rule_check function, you can simply `return not rule_note` - if no error, the `rule_note` would be an empty string and it is falsy in python.

Pay extra attention to the manual return function. You will need to make sure the other parameters are all identical (pass the test) before running the manual return. This means you need to compare all the other parameters before checking the equality of the uses.

Reply:
Addressed.